src/whir/committer/reader.rs

```diff
@@ -35,6 +47,28 @@ where
         &self,
         verifier_state: &mut VerifierState,
     ) -> ProofResult<ParsedCommitment<F, MerkleConfig::InnerDigest>>
+    where
+        VerifierState: UnitToBytes
+            + FieldToUnitDeserialize<F>
+            + UnitToField<F>
+            + DigestToUnitDeserialize<MerkleConfig>
+            + BytesToUnitDeserialize,
+    {
+        let mut batch_size_le_bytes: [u8; 4] = [0, 0, 0, 0];
+        verifier_state.fill_next_bytes(&mut batch_size_le_bytes)?;
+        let batch_size = u32::from_le_bytes(batch_size_le_bytes) as usize;
+
+        if batch_size == 1 {
```

Comment:
Can we not have a single function instead of two, seems like a lot of duplication